DOC-195 - Spring updates (#1466)
hazelcast/hz-docs: docs/modules/spring/pages/add-caching.adoc

= Add caching to Spring

As of version 3.1, Spring Framework provides support for adding caching into an existing Spring application. From version 3.2, Spring also supports JCache-compliant caching providers. 

To use Hazelcast as the cache manager in your Spring application:

. Define a `CacheManager` bean with type `HazelcastCacheManager`.
. Pass a `HazelcastInstance` object, typically as a bean.

You can use Hazelcast as a standard Spring cache manager or a JCache-compliant cache manager and deploy it on both member and client sides.

== Enable Hazelcast as a Spring cache manager

> The following declarative configuration creates a Hazelcast instance to use as a cache manager. Note that the Hazelcast instance bean is not strictly needed because Spring Boot would detect Hazelcast on the classpath and configure an instance automatically.

Reviewer: I think I see why you've put "would" but reads awk to me - either "will" or 
```suggestion
The following declarative configuration creates a Hazelcast instance to use as a cache manager. Note that the Hazelcast instance bean is not strictly needed because Spring Boot detects Hazelcast on the classpath and configures an instance automatically.
```

--
[source,java]
----

> import javax.management.MXBean;

Reviewer 2: ```suggestion
```

<cache:annotation-driven cache-manager="cacheManager" />

<hz:hazelcast id="instance">
    ...
</hz:hazelcast>

<hz:cache-manager id="hazelcastJCacheCacheManager" instance-ref="instance" name="hazelcastJCacheCacheManager"/>

<bean id="cacheManager" class="org.springframework.cache.jcache.JCacheCacheManager">
    <constructor-arg ref="hazelcastJCacheCacheManager" />
</bean>
----

> == Reference the cache manager

Reviewer 2: ```suggestion
== Reference the Hazelcast instance in the cache manager
```

    </hz:properties>
</hz:cache-manager>
----
--
====

> // Should we have a Java equivalent here?

Author: @TomaszGaweda ?

Reviewer 2: Sorry for late reply - this tip is XML-specific, as in Java config user simply passes reference
